cmd/kubeadm/app/preflight/checks.go

```diff
@@ -113,31 +113,44 @@ func (irc IsRootCheck) Check() (warnings, errors []error) {
 // DirAvailableCheck checks if the given directory either does not exist, or
 // is empty.
 type DirAvailableCheck struct {
-	path string
+	Path string
 }
 
 func (dac DirAvailableCheck) Check() (warnings, errors []error) {
 	errors = []error{}
 	// If it doesn't exist we are good:
-	if _, err := os.Stat(dac.path); os.IsNotExist(err) {
+	if _, err := os.Stat(dac.Path); os.IsNotExist(err) {
 		return nil, nil
 	}
 
-	f, err := os.Open(dac.path)
+	f, err := os.Open(dac.Path)
 	if err != nil {
-		errors = append(errors, fmt.Errorf("unable to check if %s is empty: %s", dac.path, err))
+		errors = append(errors, fmt.Errorf("unable to check if %s is empty: %s", dac.Path, err))
 		return nil, errors
 	}
 	defer f.Close()
 
 	_, err = f.Readdirnames(1)
 	if err != io.EOF {
-		errors = append(errors, fmt.Errorf("%s is not empty", dac.path))
+		errors = append(errors, fmt.Errorf("%s is not empty", dac.Path))
 	}
 
 	return nil, errors
 }
 
+// FileAvailableCheck checks that the given file does not already exist.
+type FileAvailableCheck struct {
+	path string
+}
+
+func (fac FileAvailableCheck) Check() (warnings, errors []error) {
+	errors = []error{}
+	if _, err := os.Stat(fac.path); err == nil {
```

Comment:
os.Stat actually throws an error if the file does not exist, so no error here means it does exist and thus we should fail the check. (alt example can be found in DirAvailableCheck)